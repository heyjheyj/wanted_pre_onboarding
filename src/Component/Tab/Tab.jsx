import React, { useState } from "react";
import styles from "./Tab.module.css";

const Tab = props => {
  const [current, setCurrent] = useState(0);
  const tabs = ["ONE", "TWO", "THREE"];

  const selected = {
    background: "#431cc6",
    color: "white",
    transition: "background 0.5s ease"
  };

  const changeTab = order => setCurrent(order);

  return (
    <div className={styles.container}>
      <h3 className={styles.title}>Tab</h3>
      <ul className={styles.tabLists}>
        {tabs.map((tab, index) =>
          <li
            key={index}
            className={styles.tab}
            data-testid={`test-tab${index + 1}`}
            onClick={() => changeTab(index)}
            style={current === index ? selected : {}}
          >
            {`Tab${index + 1}`}
          </li>
        )}
      </ul>
      <span className={styles.text} data-testid="test-text">
        Tab Menu {tabs[current]}
      </span>
    </div>
  );
};

export default Tab;
