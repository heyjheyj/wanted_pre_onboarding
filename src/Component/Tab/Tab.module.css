.container {
    width: 80%;
    height: 200px;
    border: 1px solid grey;
    margin-top: 10px;
    display: flex;
    flex-direction: row;
    position: relative;
    border-radius: 10px;
}

.text {
    position: absolute;
    top: 120px;
    left: 40%;
}

.title {
    position: absolute;
    top: 5px;
    left: 5px;
}

.tabLists {
    position: absolute;
    top: 10px;
    right: 0;
    width: 90%;
    height: 50px;
    background-color: #bbb;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
}

.tab {
    padding-left: 5px;
    width: 31%;
    color: grey;
    display: flex;
    align-items: center;
    cursor: pointer;
}
