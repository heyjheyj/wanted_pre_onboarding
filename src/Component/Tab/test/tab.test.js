import React from "react";
import Tab from "../Tab";
import TestRenderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Tab", () => {
  const component = TestRenderer.create(<Tab />);

  it("renders", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("Tab Component", () => {
    let text;
    beforeEach(() => {
      render(<Tab />);
      text = screen.getByTestId("test-text");
    });

    it("changes the background color in tab1 and text in body when tab1 is clicked", () => {
      let tab1 = screen.getByTestId("test-tab1");

      userEvent.click(tab1);
      expect(tab1.style.background).toBe("rgb(67, 28, 198)");
      expect(tab1.style.color).toBe("white");
      expect(text.textContent).toBe("Tab Menu ONE");
    });

    it("changes the background color in tab2 and text in body when tab2 is clicked", () => {
      let tab2 = screen.getByTestId("test-tab2");

      userEvent.click(tab2);
      expect(tab2.style.background).toBe("rgb(67, 28, 198)");
      expect(tab2.style.color).toBe("white");
      expect(text.textContent).toBe("Tab Menu TWO");
    });

    it("changes the background color in tab3 and text in body when tab3 is clicked", () => {
      let tab3 = screen.getByText("Tab3");

      userEvent.click(tab3);
      expect(tab3.style.background).toBe("rgb(67, 28, 198)");
      expect(tab3.style.color).toBe("white");
      expect(text.textContent).toBe("Tab Menu THREE");
    });
  });
});
