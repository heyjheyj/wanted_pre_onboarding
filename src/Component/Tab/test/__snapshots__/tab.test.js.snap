// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders 1`] = `
<div
  className="container"
>
  <h3
    className="title"
  >
    Tab
  </h3>
  <ul
    className="tabLists"
  >
    <li
      className="tab"
      data-testid="test-tab1"
      onClick={[Function]}
      style={
        Object {
          "background": "#431cc6",
          "color": "white",
          "transition": "background 0.5s ease",
        }
      }
    >
      Tab1
    </li>
    <li
      className="tab"
      data-testid="test-tab2"
      onClick={[Function]}
      style={Object {}}
    >
      Tab2
    </li>
    <li
      className="tab"
      data-testid="test-tab3"
      onClick={[Function]}
      style={Object {}}
    >
      Tab3
    </li>
  </ul>
  <span
    className="text"
    data-testid="test-text"
  >
    Tab Menu 
    ONE
  </span>
</div>
`;
